example/provider-extensions/seed/create-seed.sh

```diff
@@ -0,0 +1,94 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses~LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+usage() {
+  echo "Usage:"
+  echo "> create-seed.sh [ -h | <garden-kubeconfig> <seed-kubeconfig> <seed-name> ]"
+  echo
+  echo ">> For example: create-seed.sh ~/.kube/garden-kubeconfig.yaml ~/.kube/kubeconfig.yaml provider-extensions"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 3 ]; then
+  usage
+fi
+
+garden_kubeconfig=$1
+seed_kubeconfig=$2
+seed_name=$3
+
+temp_shoot_info=$(mktemp)
+cleanup-shoot-info() {
+  rm -f "$temp_shoot_info"
+}
+trap cleanup-shoot-info EXIT
+
+default-if-initial() {
+  local var=$1
+  local file=$2
+  local yqArg=$3
+  local prefix=$4
+
+  if [[  $var  == "" ]] || [[  $var  == "null" ]]; then
+    echo "${prefix}$(yq "${yqArg}" "$file")"
+  else
+    echo "$var"
+  fi
+}
+
+ingress_domain=$(yq -e '.ingressDomain' "$SCRIPT_DIR"/seed-config.yaml)
+
+registry_domain=$(yq '.registryDomain' "$SCRIPT_DIR"/seed-config.yaml)
+pods_cidr=$(yq '.podNetwork' "$SCRIPT_DIR"/seed-config.yaml)
+nodes_cidr=$(yq '.nodeNetwork' "$SCRIPT_DIR"/seed-config.yaml)
+services_cidr=$(yq '.serviceNetwork' "$SCRIPT_DIR"/seed-config.yaml)
+region=$(yq '.region' "$SCRIPT_DIR"/seed-config.yaml)
+type=$(yq '.provider' "$SCRIPT_DIR"/seed-config.yaml)
+internal_dns_secret=$(yq -e '.global.internalDomain.domain' "$SCRIPT_DIR"/../../gardener-local/controlplane/extensions-config/values.yaml | sed 's/\./-/g' | sed 's/^/internal-domain-/')
+dns_provider_type=$(yq -e '.global.internalDomain.provider' "$SCRIPT_DIR"/../../gardener-local/controlplane/extensions-config/values.yaml)
+
+if [[ $(yq '.useGardenerShootInfo' "$SCRIPT_DIR"/seed-config.yaml) == "true" ]]; then
+  echo "Getting config from shoot"
+  kubectl get configmaps -n kube-system shoot-info --kubeconfig "$seed_kubeconfig" -o yaml > "$temp_shoot_info"
+
+  registry_domain=$(default-if-initial "$registry_domain" "$temp_shoot_info" ".data.domain" "reg.")
+  pods_cidr=$(default-if-initial "$pods_cidr" "$temp_shoot_info" ".data.podNetwork")
+  nodes_cidr=$(default-if-initial "$nodes_cidr" "$temp_shoot_info" ".data.nodeNetwork")
+  services_cidr=$(default-if-initial "$services_cidr" "$temp_shoot_info" ".data.serviceNetwork")
+  region=$(default-if-initial "$region" "$temp_shoot_info" ".data.region")
+  type=$(default-if-initial "$type" "$temp_shoot_info" ".data.provider")
+fi
+
+echo "Skaffolding seed"
```

Comment:
I'm not sure if we should more or less duplicate the seed API here (by using env vars). Everytime the API is enhanced, we might need to change it here as well, e.g. `spec.provider.zones` is missing today. Isn't it less effort if we mimic the seed spec in a `values` file? Then we at least don't need to modify this file, instead developer can just enhance their local `values` file.

Reply:
There is a [gardenlet values.yaml](https://github.com/oliver-goetz/gardener/blob/enh/gardener-local-with-extensions/example/gardener-local/gardenlet/values-provider-extensions.yaml) which is used by skaffold. General things could be added here, as it is done for the other local setup configurations too.
These environment variables overwrite certain variables like DNS configuration or CIDR ranges, which depend on the concrete cluster you use. 
I'm following the approach with an own config file for the seed, because when you start this scenario it looks easier when you need to configure like 10 values (or 3 values in case you are using an Gardener shoot as seed) instead of an entire gardenlet values.yaml.

Reply:
I wonder how developers can adjust their setup, e.g. by adding specific zones for one seed, w/o touching any checked in files. Is that already possible? 👀

Reply:
Please see https://github.com/gardener/gardener/pull/6678#discussion_r1037067198